docs/user/user/metrics/rate.md

```diff
@@ -0,0 +1,129 @@
+# Rate
+
+Used to count how often something happens relative to how often something else happens.
+Like how many documents use a particular CSS Property,
+or how many HTTP connections had an error.
+You can think of it like a fraction, with a numerator and a denominator.
+
+All rates start without a value.
+A rate with a numerator of 0 is valid and will be sent to ensure we capture the
+"no errors happened" or "no use counted" cases.
+
+> **IMPORTANT:** When using a rate metric, it is important to let the Glean metric do the counting.
+  Using your own variable for counting and setting the metric yourself could be problematic:
+  ping scheduling will make it difficult to ensure the metric is at the correct value at the correct time.
+  Instead, count to the numerator and denominator as you go.
+
+## Configuration
+
+Say you're adding a new rate for how often HTTP connections have errors.
+First you need to add an entry for the rate to the `metrics.yaml` file:
+
+```YAML
+network:
+  http_connection_error:
+    type: rate
+    description: >
+      How many HTTP connections error out out of the total connections made.
+    ...
+```
+
+### External Denominators
+
+If several rates share the same denominator
+(from our example above, maybe there are multiple rates per total connections made)
+then the denominator should be defined as a `counter` and shared between
+`rates` using the `denominator_metric` property:
+
+```YAML
+network:
+  http_connections:
+    type: counter
+    description: >
+      Total number of http connections made.
+    ...
+
+  http_connection_error:
+    type: rate
+    description: >
+      How many HTTP connections error out out of the total connections made.
+    denominator_metric: network.http_connections
+    ...
+```
+
+## API
+
+{{#include ../../../shared/tab_header.md}}
+
+<div data-lang="Rust" class="tab">
+
+Since a rate is two numbers, you add to each one individually:
+
+```rust
+use glean_metrics::*;
+
+if connection_had_error {
+    network::http_connection_error.add_to_numerator(1);
+}
+
+network::http_connection_error.add_to_denominator(1);
+```
+
+If the rate uses an external denominator,
+adding to the denominator must be done through the denominator's
+`counter` API:
+
+```rust
+use glean_metrics;
+
+if connection_had_error {
+    network::http_connection_error.add_to_numerator(1);
+}
+
+// network::http_connection_error has no `add_to_denominator` method.
+network::http_connections.add(1);
```

Comment:
The proposal says the API is the same whether the denominator is internal or external.  Did that change?

Reply:
Yes. To Jan-Erik and I it seemed likely (upon review) that use of `add_to_denominator` with a shared denominator would result in misuse. `network::http_connection_error.add_to_denominator` and `network::http_connection_success.add_to_denominator` do not immediately highlight that they're talking about the same denominator. But if we get rid of the API, it forces the user to use `network::http_connections.add` which should be harder to misuse.

Reply:
Yeah, it's definitely more transparent as to what's happening.  And I can't really think of the magical use case where you want to be sort of "polymorphic" over internal and external denominators.  So seems fine -- mostly just me not being up to speed since the proposal.  (I would say updating the proposal to reflect this is a non-goal -- the docs in this PR become the canonical description of what *is* -- but that's a grey area).

Reply:
I've thrown a comment in the doc anyway so that context lives on. It's a low-energy-enough ask. But I'm a firm believer that these API docs (and the developer comments, and the code) are the living product of the proposal process and should be taken as canonical once they land.

(( I noodled an explanatory note in the docs to editorialize on how it could be misused, but I couldn't make it clear so I left it out for brevity. The API of add_to_denominator doesn't even exist for external-denominator rates, so it felt weird talking about how it could be misused. ))